### DISCUSSION TITLE
Mutation and query for asset folder (GraphQL)

### DISCUSSION THREAD

**Discussion by andrewfairlie** [craftcms/cms#8972] [Ideas]
### Description
As far as I can tell there's currently no way to create or find a new asset folder via GraphQL.

Ideally you'd be able to do something like this...

```
mutation {
  save_asset_folder(sourceId:1, folderName: "example"){
    id
  }
}

query {
  asset_folder(sourceId:1, folderName: "example"){
    id 
  }
}
```

This would allow the frontend to create and access specific folders for specific users to keep user-generated assets neat.

**Comment by andris-sevcenko** [comment]
Yeah sometime down the road we'll add non-element operations to the core GraphQL API, but that's unlikely even for 4.0.
